### PR TITLE
Responsive favorite security

### PR DESCRIPTION
**Pull request by remy-rm**
add security and responsive

### CODE REVIEW

**Review by ArPimentel — APPROVED** [review]

File: src/app/component/recipes-display/recipes-display.component.html [hunk 1]
```diff
@@ -25,43 +42,100 @@ <h2 class="allrecipes--title">appetizers</h2>
       <h2 class="allrecipes--title">breakfasts</h2>
       <div class="allrecipes_container">
         <div class="infinite-scroll" [style.transform]="'translateX(' + translateValueBreakfasts + 'px)'">
-          <div class="allrecipes_container--card" [style.transform]="'translateX(' + (translateValueBreakfastsReverse) + 'px)'">
-            <app-card-recipes *ngFor="let recipe of recipes.breakfasts; let i = index" [recipe]="recipe" [ngClass]="[(isPageLoaded ? '' : 'card_animation--' + (i + 1))]"></app-card-recipes>
+          <div
+            class="allrecipes_container--card"
+            [style.transform]="'translateX(' + translateValueBreakfastsReverse + 'px)'"
+          >
+            <app-card-recipes
```
Comment (Benja-san) [inline src/app/component/recipes-display/recipes-display.component.html:49]: same comment as upside

File: src/app/component/recipes-display/recipes-display.component.html [hunk 2]
```diff
@@ -25,43 +42,100 @@ <h2 class="allrecipes--title">appetizers</h2>
...
-      <div class="allrecipes_container" >
+      <div class="allrecipes_container">
         <div class="infinite-scroll" [style.transform]="'translateX(' + translateValueMainDishes + 'px)'">
-          <div class="allrecipes_container--card" [style.transform]="'translateX(' + (translateValueMainDishesReverse) + 'px)'">
-            <app-card-recipes *ngFor="let recipe of recipes.mainDishes; let i = index" [recipe]="recipe" [ngClass]="[(isPageLoaded ? '' : 'card_animation--' + (i + 1))]"></app-card-recipes>
+          <div
+            class="allrecipes_container--card"
+            [style.transform]="'translateX(' + translateValueMainDishesReverse + 'px)'"
+          >
+            <app-card-recipes
```
Comment (Benja-san) [inline src/app/component/recipes-display/recipes-display.component.html:83]: you know what to do

File: src/app/component/recipes-display/recipes-display.component.html [hunk 3]
```diff
@@ -25,43 +42,100 @@ <h2 class="allrecipes--title">appetizers</h2>
...
     </section>
 
     <section *ngIf="recipes.sideDishes.length > 0">
       <h2 class="allrecipes--title">side dishes</h2>
       <div class="allrecipes_container">
-        <div class="infinite-scroll"  [style.transform]="'translateX(' + translateValueSideDishes + 'px)'" >
-          <div class="allrecipes_container--card infinite-scroll" [style.transform]="'translateX(' + (translateValueSideDishesReverse) + 'px)'">
-            <app-card-recipes *ngFor="let recipe of recipes.sideDishes; let i = index" [recipe]="recipe" [ngClass]="[(isPageLoaded ? '' : 'card_animation--' + (i + 1))]"></app-card-recipes>
+        <div class="infinite-scroll" [style.transform]="'translateX(' + translateValueSideDishes + 'px)'">
+          <div
```
Comment (Benja-san) [inline src/app/component/recipes-display/recipes-display.component.html:113]: here as well

File: src/app/component/recipes-display/recipes-display.component.ts [hunk 4]
```diff
@@ -92,41 +95,39 @@ export class RecipesdisplayComponent implements OnInit, OnChanges {
     } else if (recipe === 'sideDishes') {
       recipeList = this.recipes.sideDishes;
     } else {
-      console.error("Invalid recipe category");
+      console.error('Invalid recipe category');
```
Comment (Benja-san) [inline src/app/component/recipes-display/recipes-display.component.ts:98]: remove console logs

File: src/app/component/recipes-display/recipes-display.component.ts [hunk 5]
```diff
@@ -92,41 +95,39 @@ export class RecipesdisplayComponent implements OnInit, OnChanges {
...
 
     if (direction > 0) {
       const lastCard = recipeList[recipeList.length - 1];
       recipeList.pop();
-      this.changeValue(recipe, -275);
+      this.changeValue(recipe, this.translateValue);
       recipeList.unshift(lastCard);
-    } else { // Scroll left
+    } else {
+      // Scroll left
```
Comment (Benja-san) [inline src/app/component/recipes-display/recipes-display.component.ts:108]: remove dead code

File: src/app/pages/favorites/favorites.component.ts [hunk 6]
```diff
@@ -22,16 +22,12 @@ export class FavoritesComponent implements OnInit, OnDestroy {
     });
   }
 
+  loadFavoriteRecipes(): void {
+    this.userService.loadFavoriteRecipes();
+    this.favoriteRecipes = this.userService.getFavoriteRecipes();
+  }
   ngOnDestroy(): void {
     this.favoriteUpdateSubscription.unsubscribe();
+    this.recipesService.resetParam(); // reset param when component is destroyed
```
Comment (Benja-san) [inline src/app/pages/favorites/favorites.component.ts:31]: remove dead code

File: src/app/pages/homepage/homepage.component.html [hunk 7]
```diff
@@ -1,6 +1,7 @@
 <div class="page-container">
   <div class="contents">
-    <img class="contents_img" src="/assets/logo/main-logo.png" alt="">
-    <a class="button" routerLink="recipes"> Welcome to Chef's Compass</a>
-</div>
+    <!--  <img class="contents_img" src="/assets/logo/main-logo.png" alt="" />-->
```
Comment (Benja-san) [inline src/app/pages/homepage/homepage.component.html:3]: remove dead code

File: src/app/pages/single-recipe/single-recipe.component.ts [hunk 8]
```diff
@@ -49,10 +59,13 @@ export class SingleRecipeComponent implements OnInit {
             if (recipe) {
               this.recipe = recipe;
               this.isLoading = false;
+              this.favoriteSubscription = this.userService.isActive(this.recipe.id).subscribe((isActive) => {
+                this.favorite = isActive;
+                console.log(this.favorite);
```
Comment (Benja-san) [inline src/app/pages/single-recipe/single-recipe.component.ts:64]: remove console log

File: src/app/services/search/search.service.ts [hunk 9]
```diff
@@ -2,21 +2,23 @@ import { Injectable } from '@angular/core';
...
+  constructor(private http: HttpClient) {}
 
-  searchUrl = 'http://localhost:8080';
+  searchUrl = 'http://localhost:8080/recipes/';
 
   getMultipleSearch(): Observable<any[]> {
     const country = this.http.get<Country[]>(`${this.searchUrl}/countries`);
     const allergens = this.http.get<Allergen[]>(`${this.searchUrl}/allergens`);
     const diets = this.http.get<Diet[]>(`${this.searchUrl}/diets`);
+    //const firstname = this.http.get<Users[]>(`${this.searchUrl}/users`);
```
Comment (Benja-san) [inline src/app/services/search/search.service.ts:21]: remove dead code

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Review by ArPimentel — APPROVED** [review]

File: src/app/component/navbar/navbar.component.html [hunk 10]
```diff
@@ -75,13 +75,13 @@
         </button>
       </div>
```
Comment (Benja-san) [inline src/app/component/navbar/navbar.component.html:76]: li as well

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Review by Benja-san — APPROVED** [review]

**Review by ArPimentel — APPROVED** [review]

**Pull request merged**